### PR TITLE
REGISTRY-3331 - Fix for Backend throws an error when updating life cy…

### PR DESCRIPTION
**Pull request by mushthaq33**
…cle event

### CODE REVIEW

File: components/governance/org.wso2.carbon.governance.api/src/main/java/org/wso2/carbon/governance/api/util/GovernanceUtils.java [hunk 1]
```diff
@@ -1651,7 +1653,8 @@ public static void removeAspect(String path, String aspect, Registry registry)
                         || propKey.startsWith("registry.custom_lifecycle.checklist.")
                         || propKey.startsWith("registry.LC.name")
                         || propKey.startsWith("registry.lifecycle.")
-                        || propKey.startsWith("registry.Aspects")) && propKey.contains(aspect)) {
+                        || propKey.startsWith("registry.Aspects"))
+                        && propKey.contains(GovernanceConstants.DOT + aspect + GovernanceConstants.DOT)) {
```
Comment (thushara35) [inline components/governance/org.wso2.carbon.governance.api/src/main/java/org/wso2/carbon/governance/api/util/GovernanceUtils.java]: how about moving these conditions to separate method.

Comment (mushthaq33) [inline components/governance/org.wso2.carbon.governance.api/src/main/java/org/wso2/carbon/governance/api/util/GovernanceUtils.java]: I have fixed this in the following commit - https://github.com/wso2/carbon-governance/pull/254/commits/65705acf8ade61414a8367c95b33b98ae7805cfc

File: components/governance/org.wso2.carbon.governance.api/src/main/java/org/wso2/carbon/governance/api/util/GovernanceConstants.java [hunk 2]
```diff
@@ -80,6 +80,11 @@
     public static final String ENTRY_VALUE_SEPARATOR = ":";
 
     /**
+     * This is the character used in lifecycle property keys
+     */
+    public static final String DOT = ".";
```
Comment (sagara-gunathunga) [inline components/governance/org.wso2.carbon.governance.api/src/main/java/org/wso2/carbon/governance/api/util/GovernanceConstants.java]: This should be a meaningful name that describe the purpose of this variable (something like "lifecycle property separator")  not a name to describe the value.

File: components/governance/org.wso2.carbon.governance.api/src/main/java/org/wso2/carbon/governance/api/util/GovernanceUtils.java (hunk 1 above)
Comment (sagara-gunathunga) [inline components/governance/org.wso2.carbon.governance.api/src/main/java/org/wso2/carbon/governance/api/util/GovernanceUtils.java]: Also introduce proper constants, it seems some literals are repetitive here

Comment (mushthaq33) [inline components/governance/org.wso2.carbon.governance.api/src/main/java/org/wso2/carbon/governance/api/util/GovernanceUtils.java]: Thanks for pointing it out. Fixed with the following commit - https://github.com/wso2/carbon-governance/commit/a9913cb98c71985adbcc934464c3a519e1ab3b92

File: components/governance/org.wso2.carbon.governance.api/src/main/java/org/wso2/carbon/governance/api/util/GovernanceConstants.java (hunk 2 above)
Comment (mushthaq33) [inline components/governance/org.wso2.carbon.governance.api/src/main/java/org/wso2/carbon/governance/api/util/GovernanceConstants.java]: Thanks for pointing it out. Fixed with the following commit - a9913cb

**Pull request merged**